signac/common/config.py

```diff
@@ -72,35 +93,52 @@ def read_config_file(filename):
     return config
 
 
-def get_config(infile=None, configspec=None, *args, **kwargs):
-    """Get configuration from a file."""
-    if configspec is None:
-        configspec = cfg.split("\n")
-    return Config(infile, configspec=configspec, *args, **kwargs)
-
-
 def load_config(root=None, local=False):
-    """Load configuration, searching upward from a root path."""
+    """Load configuration, searching upward from a root path if desired.
+
+    Parameters
+    ----------
+    root : str
+        The path from which to begin searching for config files.
+    local : bool, optional
+        If ``True``, only search in the provided directory and do not traverse
+        upwards through the filesystem (Default value: False).
+
+    Returns
+    --------
+    (str, :class:`Config`)
+        The directory in which a config was found and the corresponding
+        configuration. If no config was found, will return ``None`` for the
+        directory and an empty config object.
+    """
     if root is None:
         root = os.getcwd()
     config = Config(configspec=cfg.split("\n"))
+
     if local:
-        for fn in _search_local(root):
-            tmp = read_config_file(fn)
-            config.merge(tmp)
-            if "project" in tmp:
-                config["project_dir"] = os.path.dirname(fn)
-                break
+        # Local searches cannot proceed up the tree.
+        search_func = _search_local
     else:
-        for fn in search_standard_dirs():
+        # For non-local searches we grab the user's global config file first.
+        for fn in _search_standard_dirs():
             config.merge(read_config_file(fn))
-        for fn in _search_tree(root):
-            tmp = read_config_file(fn)
-            config.merge(tmp)
-            if "project" in tmp:
-                config["project_dir"] = os.path.dirname(fn)
-                break
-    return config
+        search_func = _search_tree
+
+    root_dir = None
+    for fn in search_func(root):
+        tmp = read_config_file(fn)
+        config.merge(tmp)
+        # Once a valid config file is found, we cease looking any further, i.e.
+        # we assume that the first directory with a valid config file is the
+        # project root.
+        # TODO: Is this a sufficiently portable solution? We could use pathlib's
+        # `.paths` attribute, but that has significant performance
+        # implications: this takes ~200 ns, whereas
+        # os.path.join(*(pathlib.PosixPath(fn).parts[:-2]))
+        # takes ~6 us, almost a 20x slowdown.
```

Comment:
I'm fine with this. Just rewrite the TODO as an explanation of behavior and design choice, rather than as a question.